### ISSUE TITLE
settimeout support

### ISSUE THREAD

**Issue by olueiro** [daurnimator/lua-http#81]
I intend to use this library for my project. Is there any function to control time intervals?

```lua
local function cb(arg1)
 -- async stuff goes here...
end

local ms = 1500
settimeout(ms, cb, "arg1")
```

Thanks in advance!

**Comment by daurnimator** [comment]
Almost all operations take a timeout as a parameter.
Possible duplicate of #76?

**Comment by olueiro** [comment]
sorry,

The related issue does not seem to meet my doubt. Where can I find an example?

**Comment by daurnimator** [comment]
> The related issue does not seem to meet my doubt. Where can I find an example?

What are you trying to do?
e.g. `:go(1)` allows it to take one second.
https://daurnimator.github.io/lua-http/0.2/#http.request:timeout

**Comment by olueiro** [comment]
I need a timer (unblocking) that runs every hour to execute a command via `io.popen`
equivalent to [ngx.timer.at](https://github.com/openresty/lua-nginx-module#ngxtimerat)
Is this possible in lua-http or do I need to use `crontab`?

**Comment by daurnimator** [comment]
lua-http works together with cqueues for scheduling.
Use e.g.

```lua
while true do
    cqueues.sleep(60*60) -- sleep for 1 hour
    local myrequest = http_request.new_from_uri("http://foo.com/")
    local headers, stream = myrequest:go(1)
    if not headers then
        -- handle error
    else
        -- handle success
        stream:shutdown()
    end
end
```

The above can be run within a cqueues managed coroutine if you need cooperative multi-tasking.

**Comment by RussellHaley** [comment]
I have some examples here: https://github.com/RussellHaley/lua-http-endpoints

Exb_client.lua (DebugInput) and exb_server.lua (PollClients)both have timed routines. Also there is filewatcher code as per cqueues authors instruction. I don't remeber the final state of the code (working/not working. It did at one point!)
